firmware/shared/util/peripheral.h

```diff
@@ -0,0 +1,17 @@
+/// @author Blake Freer
+/// @date 2023-11-08
+
+#ifndef SHARED_UTIL_PERIPHERAL_H_
+#define SHARED_UTIL_PERIPHERAL_H_
+
+#include "shared/util/noncopyable.h"
+
+namespace shared::util {
+
+class Peripheral {//: private Noncopyable {
```

Comment:
Might be missing it, but what is the purpose of this

Reply:
It's generic properties (like move semantics) that are inherited by all peripherals. Currently it is empty but I want to keep it available for the future. Since we don't reference anything bu the parent Peripheral class, this does not break the "avoiding abstraction" goal